File: notification-service/src/main/java/ru/javavlsu/kb/notificationservice/kafka/KafkaProducer.java
package ru.javavlsu.kb.notificationservice.kafka;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

@Component
public class KafkaProducer {

    @Value("${notifications.topic.name}")
    private String notificationsTopic;
    private final ObjectMapper objectMapper;
    private final KafkaTemplate<String, String> kafkaTemplate;
    private final Logger log = LoggerFactory.getLogger(KafkaProducer.class);

    @Autowired
    public KafkaProducer(ObjectMapper objectMapper, KafkaTemplate<String, String> kafkaTemplate) {
        this.objectMapper = objectMapper;
        this.kafkaTemplate = kafkaTemplate;
    }

    public void sendTokenStatusMessage(String token) {
        String message = "Token {token=" + token + "} is inactive due to device unavailability";
        ProducerRecord<String, String> record = new ProducerRecord<>(notificationsTopic, message);
        record.headers().add("type", MessageType.RESPONSE.name().getBytes());
        kafkaTemplate.send(record);
        log.info(message);
    }
}

File: notification-service/src/main/kotlin/ru/javavlsu/kb/notificationservice/kafka/KafkaProducer.kt
package ru.javavlsu.kb.notificationservice.kafka

import org.apache.kafka.clients.producer.ProducerRecord
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component

@Component
class KafkaProducer constructor(
    @Value("\${notifications.topic.name}")
    private val notificationsTopic: String,
    private val kafkaTemplate: KafkaTemplate<String, String>
) {
    private val log = LoggerFactory.getLogger(KafkaProducer::class.java)

    fun sendTokenStatusMessage(token: String) {
        val message = "Token {token=$token} is inactive due to device unavailability"
        val record = ProducerRecord<String, String>(notificationsTopic, message)
        record.headers().add("type", MessageType.RESPONSE.name.toByteArray())
        kafkaTemplate.send(record)
        log.info(message)
    }
}